### PR TITLE
[BIOMAGE-1547] Reduce the number of default pipeline pods in staging and in production

### PR DESCRIPTION
**Pull request by ivababukova**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1547

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -229,7 +229,7 @@ jobs:
             SANDBOX_ID="STAGING_SANDBOX_ID"
             CHART_REF="STAGING_CHART_REF"
             KUBERNETES_ENV="staging"
-            REPLICA_COUNT="2"
+            REPLICA_COUNT="1"
```
Comment (kafkasl) [inline .github/workflows/ci.yaml:232]: we can change this to 0 (it's the staging envs and they will be scaled back anyway)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -210,7 +210,7 @@ jobs:
             SANDBOX_ID="default"
             CHART_REF="$GITHUB_SHA"
             KUBERNETES_ENV="production"
-            REPLICA_COUNT="4"
+            REPLICA_COUNT="1"
```
Comment (kafkasl) [inline .github/workflows/ci.yaml:213]: if we need to reduce this, I would suggest at least 3, running the pipeline needs 2 pods, so it'd be great to have a third one if another user is working too (I would keep it to 4)

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -220,7 +220,7 @@ jobs:
             SANDBOX_ID="default"
             CHART_REF="$GITHUB_SHA"
             KUBERNETES_ENV="staging"
-            REPLICA_COUNT="2"
+            REPLICA_COUNT="1"
```
Comment (kafkasl) [inline .github/workflows/ci.yaml:223]: we'll have to wait a bit of extra time when running the pipeline from gem2s (if it's slower than 3/4 minutes) but should be ok, we can decrease and see if it bothers people

**Comment by ivababukova** [comment]
All fair points there, I have addressed them

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (ivababukova) [reply .github/workflows/ci.yaml:223]: Let's try with 1 and see if it's ok. If it isn't, we will increase it back to 2.

**Review by kafkasl — APPROVED** [review]

**Pull request merged**